.travis.yml

```diff
@@ -119,19 +113,21 @@ before_install:
     - export PATH=$(echo $PATH | tr ':' "\n" | sed '/\/opt\/python/d' | tr "\n" ":" | sed "s|::|:|g")
     # Install the non-python dependencies: fftw, libav, eigen
     - if [[ $TRAVIS_OS_NAME == "linux" ]]; then sudo -H apt-get -qq update; sudo -H apt-get install -y python-dev libfftw3-dev; fi
+    # py3.9 system doesn't seem to have blas, lapack.
+    - if [[ $TRAVIS_OS_NAME == "linux" && $TRAVIS_PYTHON_VERSION == 3.9 ]]; then sudo -H apt-get install -y libblas-dev liblapack-dev; fi
     # Don't install eigen on 3.7 so we test the ability of setup.py to install it for us.
     - if [[ $TRAVIS_OS_NAME == "linux" && $TRAVIS_PYTHON_VERSION < 3.7 ]]; then sudo -H apt-get install -y libeigen3-dev; fi
     # Shouldn't usually need this, but occasionally might.
     #- if [[ $TRAVIS_OS_NAME == "osx" ]]; then brew update; fi
-    - if [[ $TRAVIS_OS_NAME == "osx" ]]; then brew install fftw wget; fi
-    - if [[ $TRAVIS_OS_NAME == "osx" && $TRAVIS_PYTHON_VERSION < 3.7 ]]; then brew install eigen; brew link --overwrite eigen; fi
-    - if [[ $TRAVIS_OS_NAME == "osx" ]]; then brew cask install gfortran; fi
-    - if [[ $TRAVIS_OS_NAME == "osx" ]]; then brew upgrade wget; fi
-    - if [[ $TRAVIS_OS_NAME == "osx" ]]; then brew link --overwrite fftw gcc wget; fi
+    - if [[ $TRAVIS_OS_NAME == "osx" ]]; then brew install fftw wget || true; fi
```

Comment:
brew returns an error code if all packages are already installed and current.  It took me a while to figure this out, but this was the source of intermittent errors in Mac installations.  If the brew install command decided there was nothing to do, it returned an "error".  Travis then declares the installation broken and says it failed.
So the apparently bespoke solution I found on some brew issue page is to add this "|| true" at the end so that if the brew command errors, the shell still returns success and Travis goes on to the next stage.  If it really was an error, we'll discover it eventually.  (I think.)